src/coreComponents/constitutive/fluid/MultiFluidBase.hpp

```diff
@@ -339,6 +478,322 @@ class MultiFluidBase : public ConstitutiveBase
 
 };
 
+template< typename OUT_ARRAY >
+GEOSX_HOST_DEVICE
+void
+MultiFluidBase::KernelWrapper::
+  convertToMoleFractions( arraySlice1d< real64 const, compflow::USD_COMP - 1 > const composition,
```

Comment:
If i understand this correctly, there are two version here, one returning derivs and one not.  I might have one call the other, to condense the actual implementation to one spot.  You could call the one with derivs from a non-deriv version with a junk deriv array, and then just discard the extra data.  It would look cleaner, at the cost of a negligeable overhead when you just want values and not derivs.  Just a suggestion.  Your preference.